File: src/main/python/QuestionDeck.py
from collections import deque

class QuestionDeck:
    def __init__(self, number_of_questions: int):
        # Usamos deque para simular LinkedList y permitir popleft() eficiente
        self._pop_questions = deque()
        self._science_questions = deque()
        self._sports_questions = deque()
        self._rock_questions = deque()

        for i in range(number_of_questions):
            self._pop_questions.append(f"Pop Question {i}")
            self._science_questions.append(f"Science Question {i}")
            self._sports_questions.append(f"Sports Question {i}")
            self._rock_questions.append(f"Rock Question {i}")

    def get_category(self, place: int) -> str:
        # Lógica de categorías basada en el módulo de la posición
        remainder = place % 4
        if remainder == 0:
            return "Pop"
        if remainder == 1:
            return "Science"
        if remainder == 2:
            return "Sports"
        return "Rock"

    def ask_question(self, place: int) -> str:
        category = self.get_category(place)
        print(f"The category is {category}")

        # Simulación del switch de Java usando un diccionario o if/elif
        if category == "Pop":
            return self._pop_questions.popleft()
        elif category == "Science":
            return self._science_questions.popleft()
        elif category == "Sports":
            return self._sports_questions.popleft()
        else:
            return self._rock_questions.popleft()

File: src/main/java/trivia/QuestionDeck.java
package trivia;

import java.util.LinkedList;

public class QuestionDeck {
    private final LinkedList<String> popQuestions = new LinkedList<>();
    private final LinkedList<String> scienceQuestions = new LinkedList<>();
    private final LinkedList<String> sportsQuestions = new LinkedList<>();
    private final LinkedList<String> rockQuestions = new LinkedList<>();

    public QuestionDeck(int numberOfQuestions) {
        for (int i = 0; i < numberOfQuestions; i++) {
            popQuestions.addLast("Pop Question " + i);
            scienceQuestions.addLast("Science Question " + i);
            sportsQuestions.addLast("Sports Question " + i);
            rockQuestions.addLast("Rock Question " + i);
        }
    }

    public String getCategory(int place) {
        if (place % 4 == 0) return "Pop";
        if (place % 4 == 1) return "Science";
        if (place % 4 == 2) return "Sports";
        return "Rock";
    }

    public String askQuestion(int place) {
        String category = getCategory(place);
        System.out.println("The category is " + category);
        
        switch (category) {
            case "Pop": return popQuestions.removeFirst();
            case "Science": return scienceQuestions.removeFirst();
            case "Sports": return sportsQuestions.removeFirst();
            default: return rockQuestions.removeFirst();
        }
    }
}